runhouse/resources/hardware/cluster.py

```diff
@@ -842,6 +858,25 @@ def disconnect(self):
         if self._rpc_tunnel:
             self._rpc_tunnel.stop()
 
+    def _sync_across_all_nodes(self, _rh_install_url):
+        loop = asyncio.new_event_loop()
+        executor = ProcessPoolExecutor()
```

Comment:
Do we need to also call: `asyncio.set_event_loop(loop)`?

According to the documentation:
```
If there’s need to set this loop as the event loop for the current context, [set_event_loop()](https://python.readthedocs.io/en/latest/library/asyncio-eventloops.html#asyncio.set_event_loop) must be called explicitly.
```
Source:
https://python.readthedocs.io/en/latest/library/asyncio-eventloops.html#asyncio.AbstractEventLoopPolicy.new_event_loop

If the existing code works, we can probably skip it for now, but it would be good to be aware of it in case of any exceptions or weird behavior that we may face in the future.

Reply:
good point, my understanding is that since this loop's usage is confined to this function we don't necessarily need to set it, but if we do then we should also close the loop as you added in your other comment. 

Seems like it's the recommended way to do it like this, so I'll update it!